### DISCUSSION TITLE
Issue with NQ.bird model

### DISCUSSION THREAD

**Discussion by lsjermiin** [iqtree/iqtree2#198] [Q&A] [answered]
Hi,

On October 12 2023, I reported a fault (#145) involving the NQ.bird model. I wrote:

I am running experimental analysis using the following commands:

iqtree2 -s 488_AA_supermatrix.fst -st AA -m MF --mset NQ.bird -mrate E,I,G,*G,I+G,R,*R,I+R,H,*H -cmax 15 -mfreq FU,FO,F -mtree -T 16 -merit BIC -safe

For some reason it crashes shortly after starting the test of the NQ.bird+H2 model.

I repeated the analysis, and the crash happened again.

Has this matter been resolved? I'd like to use the NQ. models, but the bug prevented me from doing it.

All the best,

Lars

**Comment by bqminh** [comment, marked as answer]
We haven't tested non-reversible models with the GHOST mixture model. So I have to say it doesn't work. An error message could have been printed instead of this. So this combination is just not supported. In fact, such a combined model is not easy to implement. Technically, we are working on rooted trees, and a mixture of branch lengths. The number of branches for each mixture class is one higher than the unrooted trees (under reversible models). So quite some implementation effort has to spend in order to make it work, then testing etc... (literally we would have to pay someone to do this work).

But: we can consider it if you can show that it matters...

**Reply by roblanf** [reply]
@bqminh can we put in an error message when people try this combination (and maybe other combinations with NQ that we have not yet implemented and tested properly)?

**Comment by lsjermiin** [comment]
This is a very good idea.

Regarding testing the different combinations of substitution models and rate heterogeneity across sites models, do you have a preferred procedure already in place or in mind? I’m contemplating testing some of these combinations.

From: roblanf ***@***.***>
Date: Sunday, 19 May 2024 at 00:04
To: iqtree/iqtree2 ***@***.***>
Cc: Jermiin, Lars ***@***.***>, Author ***@***.***>
Subject: Re: [iqtree/iqtree2] Issue with NQ.bird model (Discussion #198)
EXTERNAL EMAIL: This email originated outside the University of Galway. Do not open attachments or click on links unless you believe the content is safe.
RÍOMHPHOST SEACHTRACH: Níor tháinig an ríomhphost seo ó Ollscoil na Gaillimhe. Ná hoscail ceangaltáin agus ná cliceáil ar naisc mura gcreideann tú go bhfuil an t-ábhar sábháilte.


@bqminh<https://github.com/bqminh> can we put in an error message when people try this combination (and maybe other combinations with NQ that we have not yet implemented and tested properly)?

**Reply by roblanf** [reply]
@lsjermiin  - best to move this to a new discussion, to keep each discussion thread on a single topic, and make sure the discussions are a good resource for searching.

**Reply by lsjermiin** [reply]
Agreed -- you can close this discussion, but try to remember to warn users if they try to use the NQ models with the *H RHAS model.

**Reply by bqminh** [reply]
Ok I'll close now, but you can also close it.
